src/status_im2/subs/wallet_2/wallet.cljs

```diff
@@ -0,0 +1,38 @@
+(ns status-im2.subs.wallet-2.wallet
+  (:require [re-frame.core :as re-frame]
+            [clojure.string :as string]))
+
+(defn calculate-raw-balance
+  [raw-balance decimals]
+  (let [raw-number (js/parseInt raw-balance)]
+    (if (not (js/isNaN raw-number))
+      (/ raw-number (Math/pow 10 (js/parseInt decimals)))
+      0)))
+
+(defn calculate-balance
+  [address tokens]
+  (let [token  (get tokens (keyword (string/lower-case address)))
+        result (reduce
+                (fn [acc item]
+                  (let [total-value-per-token (reduce (fn [ac balances]
+                                                        (+ (calculate-raw-balance (:rawBalance balances)
+                                                                                  (:decimals item))
+                                                           ac))
+                                                      0
+                                                      (vals (:balancesPerChain item)))
+                        total-values          (* total-value-per-token
+                                                 (get-in item [:marketValuesPerCurrency :USD :price]))]
+                    (+ acc total-values)))
+                0
+                token)]
+    (.toFixed result 2)))
```

Comment:
`toFixed` returns a string, which is odd for a function named `calculate-balance`. Calculations strongly give the impression we're returning numbers. This can be problematic if we mistakenly reuse this `:wallet-2/balances` subscription to do further calculations.

I think it would be better if the formatted value to fixed notation was performed at the view layer, i.e. we drop `.toFixed` and return `result`.

Reply:
Done